### PR TITLE
refine explanation for general log

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add detailed description for `tidb_general_log`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4843
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -417,15 +417,25 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
+- To see all records of this feature in the log, query the `"GENERAL_LOG"` string. The following information is recorded:
+    - `conn`: The ID of the current session.
+    - `user`: The current session user.
+    - `schemaVersion`: The current schema version.
+    - `txnStartTS`: The timestamp at which the current transaction starts.
+    - `forUpdateTS`: In the pessimistic transactional model, `forUpdateTS` is the current timestamp of the SQL statement. When a write conflict occurs in the pessimistic transaction, TiDB retries the SQL statement currently being executed and updates this timestamp. You can configure the number of retries via [`max-retry-count`](/tidb-configuration-file.md#max-retry-count). In the optimistic transactional model, `forUpdateTS` is equivalent to `txnStartTS`.
+    - `isReadConsistency`: Indicates whether the current transactional isolation level is Read Committed (RC).
+    - `current_db`: The name of the current database.
+    - `txn_mode`: The transactional model. Value options are `OPTIMISTIC` and `PESSIMISTIC`.
```
Comment (ran-huang) [inline system-variables.md:437]:
```suggestion
    - `txn_mode`: The transactional mode. Value options are `OPTIMISTIC` and `PESSIMISTIC`.
```

**Comment by ran-huang** [comment]
LGTM

**Review by XuHuaiyu — COMMENTED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Comment by kolbe** [comment]
Fixes #3720

**Pull request merged**